python/phevaluator/evaluator_omaha.py

```diff
@@ -64,61 +68,73 @@ def evaluate_omaha_cards(*cards: int | str | Card) -> int:
-            and suit_count_hole[i] >= min_flush_count_hole
-        ):
-            flush_suit = i
-            break
+    return min(value_flush, value_noflush)
 
-    if flush_suit != -1:
-        flush_count_board = suit_count_board[flush_suit]
-        flush_count_hole = suit_count_hole[flush_suit]
 
-        suit_binary_board = 0
-        for community_card in community_cards:
-            if community_card % 4 == flush_suit:
-                suit_binary_board |= BINARIES_BY_ID[community_card]
+def _count_suits(cards: list[int]) -> list[int]:
+    suit_count = [0] * 4
+    for card in cards:
+        suit_count[card % 4] += 1
+    return suit_count
 
-        suit_binary_hole = 0
-        for hole_card in hole_cards:
-            if hole_card % 4 == flush_suit:
-                suit_binary_hole |= BINARIES_BY_ID[hole_card]
 
+def _determine_flush_suit(
+    suit_count_board: list[int], suit_count_hole: list[int]
+) -> int:
+    for i in range(4):
         if (
-            flush_count_board == min_flush_count_board
-            and flush_count_hole == min_flush_count_hole
+            suit_count_board[i] >= MIN_FLUSH_COUNT_BOARD
+            and suit_count_hole[i] >= MIN_FLUSH_COUNT_HOLE
         ):
-            value_flush = FLUSH[suit_binary_board | suit_binary_hole]
-
-        else:
-            padding = [0x0000, 0x2000, 0x6000]
-
-            suit_binary_board |= padding[COMMUNITY_CARD_COUNT - flush_count_board]
-            suit_binary_hole |= padding[HOLE_CARD_COUNT - flush_count_hole]
-
-            board_hash = hash_binary(suit_binary_board, COMMUNITY_CARD_COUNT)
-            hole_hash = hash_binary(suit_binary_hole, HOLE_CARD_COUNT)
-
-            value_flush = FLUSH_OMAHA[board_hash * 1365 + hole_hash]
-
+            return i
+    return -1
+
+
+def _determine_flush_value(
+    community_cards: list[int],
+    hole_cards: list[int],
+    suit_count_board: list[int],
+    suit_count_hole: list[int],
+    flush_suit: int,
+) -> int:
+    flush_count_board = suit_count_board[flush_suit]
+    flush_count_hole = suit_count_hole[flush_suit]
+    suit_binary_board = 0
+    for community_card in community_cards:
+        if community_card % 4 == flush_suit:
+            suit_binary_board |= BINARIES_BY_ID[community_card]
+    suit_binary_hole = 0
+    for hole_card in hole_cards:
+        if hole_card % 4 == flush_suit:
+            suit_binary_hole |= BINARIES_BY_ID[hole_card]
+    if (
+        flush_count_board == MIN_FLUSH_COUNT_BOARD
+        and flush_count_hole == MIN_FLUSH_COUNT_HOLE
+    ):
+        return FLUSH[suit_binary_board | suit_binary_hole]
+    padding = [0x0000, 0x2000, 0x6000]
+    suit_binary_board |= padding[COMMUNITY_CARD_COUNT - flush_count_board]
+    suit_binary_hole |= padding[HOLE_CARD_COUNT - flush_count_hole]
+    board_hash = hash_binary(suit_binary_board, COMMUNITY_CARD_COUNT)
+    hole_hash = hash_binary(suit_binary_hole, HOLE_CARD_COUNT)
+    return FLUSH_OMAHA[board_hash * 1365 + hole_hash]
```

Comment:
Since we are here, it would be nice to have a const declaration/definition of the magic number 1365 too.

1365 is basically the number of possible hash values of the hole cards from a flush. There are 4 hole cards, out of 13 ranks plus 2 bit padding. So the calculation is `choose(15, 4) = 1365`. If we are to choose a name for it, maybe declare it as `MAX_FLUSH_HOLE_HASH = 1365`.